config/zen/make_defs.mk

```diff
@@ -32,53 +33,22 @@
 #
 #
 
+# FLAGS specific to zen architecture are added here.
+# FLAGS that are common for all the AMD architectures are present in amd_config.mk
 
 # Declare the name of the current configuration and add it to the
 # running list of configurations included by common.mk.
 THIS_CONFIG    := zen
 #CONFIGS_INCL   += $(THIS_CONFIG)
 
-#
-# --- Determine the C compiler and related flags ---
-#
-
-# NOTE: The build system will append these variables with various
-# general-purpose/configuration-agnostic flags in common.mk. You
-# may specify additional flags here as needed.
-CPPROCFLAGS    :=
-CMISCFLAGS     :=
-CPICFLAGS      :=
-CWARNFLAGS     :=
-
-ifneq ($(DEBUG_TYPE),off)
-CDBGFLAGS      := -g
-endif
+# Include file containing common flags for all AMD architectures
+AMD_CONFIG_FILE := amd_config.mk
+AMD_CONFIG_PATH := $(BASE_SHARE_PATH)/config/zen
+-include $(AMD_CONFIG_PATH)/$(AMD_CONFIG_FILE)
 
-ifeq ($(DEBUG_TYPE),noopt)
-COPTFLAGS      := -O0
-else
-COPTFLAGS      := -O3 -fomit-frame-pointer
-endif
-
-# Flags specific to optimized kernels.
-CKOPTFLAGS     := $(COPTFLAGS)
 ifeq ($(CC_VENDOR),gcc)
-# gcc 6.0 (clang 4.0) or later:
-#CKVECFLAGS     := -mavx2 -mfpmath=sse -mfma -march=znver1
-# gcc 4.9 (clang 3.5) or later:
-# possibly add zen-specific instructions: -mclzero -madx -mrdseed -mmwaitx -msha -mxsavec -mxsaves -mclflushopt -mpopcnt
-CKVECFLAGS     := -mavx2 -mfpmath=sse -mfma -march=bdver4 -mno-fma4 -mno-tbm -mno-xop -mno-lwp
```

Comment:
The `-mno-tbm -mno-xop -mno-lwp` flags are important for testing on Travis, since it will actually be running under Intel SDE. I highly doubt that they will affect performance in any way.

Reply:
tbm, xop, lwp, fma4 are extensions which were enabled by default in older AMD architectures like bulldozer but they have been deprecated with Zen architectures. so, we don't need to explicitly use -mno-tbm -mno-xop -mno-lwp flags to disable them.

Reply:
Ah, OK. They would only be needed if using bdver4 as a fake znver1 on older gcc versions then (the updated version would require gcc 6.0+ or clang 4.0+).